org-code-javabuilder/theater/src/main/java/org/code/theater/Stage.java

```diff
@@ -1,17 +1,63 @@
 package org.code.theater;
 
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
 import java.io.FileNotFoundException;
-import org.code.media.*;
+import org.code.media.Color;
+import org.code.media.Image;
+import org.code.protocol.GlobalProtocol;
+import org.code.protocol.OutputAdapter;
 
 public class Stage {
-  /** Returns the width of the theater canvas. Right now this will always be 400 pixels. */
-  public double getWidth() {
-    return -1;
+  private BufferedImage image;
+  private final OutputAdapter outputAdapter;
+  private final GifWriter gifWriter;
+  private final ByteArrayOutputStream outputStream;
+  private java.awt.Color strokeColor;
+  private java.awt.Color fillColor;
+  private Graphics2D graphics;
+  private boolean hasPlayed;
+
+  private static final int WIDTH = 400;
+  private static final int HEIGHT = 400;
+
+  /**
+   * Initialize Stage with a default image. Stage should be initialized outside of org.code.theater
+   * using Theater.stage.
+   */
+  protected Stage() {
+    this(new BufferedImage(WIDTH, HEIGHT, BufferedImage.TYPE_INT_RGB));
+  }
+
+  /**
+   * Initialize Stage with a specific BufferedImage. Used directly for testing. Stage should be
+   * initialized outside of org.code.theater using Theater.stage.
+   *
+   * @param image
+   */
+  protected Stage(BufferedImage image) {
```

Comment:
currently we don't (we can already mock the output adapter because it's being creating by GlobalProtocol), and can mock the graphics by mocking `createGraphics` on `image`.